### ISSUE TITLE
Some footnotes disappear from LaTeX output

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#2292]
This issue is caused by commit 9fec96d32742bf753cad0a9b1f9ddff9e24f2a06

Consider the following file `aaa.rst`:

``` rst
===
AAA
===

The footnote will vanish from ``make latex`` produced file with commit 9fec96d, 
but it was there with its parent commit 885b997.


aaaaaaaa [1]_

.. [1] cccccccc

```

and file `uuu.rst`:

``` rst
===
BBB
===

uuuuuuu [#]_ 

.. [#] see  :py:func:`repr`

```

and file `index.rst`

``` rst

===
MMM
===

.. toctree::
   :maxdepth: 2

   aaa
   uuu

```

and standard  `conf.py` with 

``` python
extensions = [
#    'sphinx.ext.autodoc',
    'sphinx.ext.intersphinx',
#    'sphinx.ext.mathjax',
]

intersphinx_mapping = {'python': ('http://docs.python.org/3', None)}
```

added.

Then on `make latex` one gets in the produced file:

``` latex
\chapter{AAA}
\label{aaa::doc}\label{aaa:aaa}\label{aaa:mmm}
The footnote will vanish from \code{make latex} produced file with commit 9fec96d,
but it was there with its parent commit 885b997.

aaaaaaaa 


\chapter{BBB}
\label{uuu::doc}\label{uuu:bbb}
uuuuuuu \footnote[2]{
see  \href{http://docs.python.org/3/library/functions.html\#repr}{\code{repr()}}
}
```

Notice that the first footnote has disappeared. This does not happen with Sphinx 1.3.3 or the parent commit 885b997a258c88a248f2f2a8514f717274103127

Archive uploaded without `build/` sub-repertory for reducing size.

[disappearingfootnote.zip](https://github.com/sphinx-doc/sphinx/files/116546/disappearingfootnote.zip)

**Comment by tk0miya** [comment]
Fixed at 9ba7cc8.

Thank you for reporting.

**Comment by jfbu** [comment]
This solves it in my real use case, too. Thanks.
